content/en/docs/languages/java/configuration.md

```diff
@@ -194,240 +306,399 @@ exporters.
+| `otel.exporter.prometheus.port`    | The local port used to bind the prometheus metric server.    | `9464`    |
+| `otel.exporter.prometheus.host`    | The local address used to bind the prometheus metric server. | `0.0.0.0` |
+
+#### Programmatic customization
+
+Programmatic customization provides hooks to supplement the
+[supported properties](#environment-variables-and-system-properties) with
+[programmatic configuration](#programmatic-configuration).
+
+<!-- prettier-ignore-start -->
+<?code-excerpt "src/main/java/otel/CustomizedAutoConfiguredSdk.java"?>
+```java
+package otel;
+
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import java.util.Collections;
+
+public class CustomizedAutoConfiguredSdk {
+  public static OpenTelemetrySdk autoconfiguredSdk() {
+    return AutoConfiguredOpenTelemetrySdk.builder()
+        // Optionally customize TextMapPropagator.
+        .addPropagatorCustomizer((textMapPropagator, configProperties) -> textMapPropagator)
+        // Optionally customize Resource.
+        .addResourceCustomizer((resource, configProperties) -> resource)
+        // Optionally customize Sampler.
+        .addSamplerCustomizer((sampler, configProperties) -> sampler)
+        // Optionally customize SpanExporter.
+        .addSpanExporterCustomizer((spanExporter, configProperties) -> spanExporter)
+        // Optionally customize SpanProcessor.
+        .addSpanProcessorCustomizer((spanProcessor, configProperties) -> spanProcessor)
+        // Optionally supply additional properties.
+        .addPropertiesSupplier(Collections::emptyMap)
+        // Optionally customize ConfigProperties.
+        .addPropertiesCustomizer(configProperties -> Collections.emptyMap())
+        // Optionally customize SdkTracerProviderBuilder.
+        .addTracerProviderCustomizer((builder, configProperties) -> builder)
+        // Optionally customize SdkMeterProviderBuilder.
+        .addMeterProviderCustomizer((builder, configProperties) -> builder)
+        // Optionally customize MetricExporter.
+        .addMetricExporterCustomizer((metricExporter, configProperties) -> metricExporter)
+        // Optionally customize MetricReader.
+        .addMetricReaderCustomizer((metricReader, configProperties) -> metricReader)
+        // Optionally customize SdkLoggerProviderBuilder.
+        .addLoggerProviderCustomizer((builder, configProperties) -> builder)
+        // Optionally customize LogRecordExporter.
+        .addLogRecordExporterCustomizer((logRecordExporter, configProperties) -> logRecordExporter)
+        // Optionally customize LogRecordProcessor.
+        .addLogRecordProcessorCustomizer((processor, configProperties) -> processor)
+        .build()
+        .getOpenTelemetrySdk();
+  }
+}
+```
+<!-- prettier-ignore-end -->
+
+#### SPI (Service provider interface)
```

Comment:
link to how this can be done with agent and starter:

https://opentelemetry.io/docs/zero-code/java/agent/extensions/
https://opentelemetry.io/docs/zero-code/java/spring-boot-starter/sdk-configuration/#programmatic-configuration

Reply:
Adding a link up in the "Programmatic customization" section, which is a better conceptual match.